geopandas/tests/test_cmapplotting.py

```diff
@@ -0,0 +1,271 @@
+        from geopandas.plotting import plot_point_collection
+        fig, ax = plt.subplots()
+        coll = plot_point_collection(ax, self.points, self.values)
+        fig.canvas.draw_idle()
+        actual_lim = coll.get_clim()
+        assert actual_lim == (0,9)
+        ax.cla()
+
+        fig, ax = plt.subplots()
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.BoundaryNorm([1,3,6,8], cmap.N)
+        coll = plot_point_collection(ax, self.points, self.values, norm = norm)
+        fig.canvas.draw_idle()
+        actual_lim = coll.get_clim()
+        assert actual_lim == (1,8)
+        ax.cla()
+
+    def test_subset_customnorm(self):
+        from geopandas.plotting import plot_point_collection
+        fig, ax = plt.subplots()
+
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.BoundaryNorm([1,2,4], cmap.N)
+        coll = plot_point_collection(ax, self.points[0:3], self.values[0:3], norm = norm)
+        fig.canvas.draw_idle()
+
+        cmap = plt.get_cmap()
+        expected_color = cmap(np.arange(self.N) / (self.N - 1))
+
+        np.testing.assert_array_equal(coll.get_facecolors()[0], expected_color[0])
+        np.testing.assert_array_equal(coll.get_facecolors()[-1], expected_color[-1])
+
+        ax.cla()
+
+    def test_subset_minmax(self):
+        from geopandas.plotting import plot_point_collection
+        fig, ax = plt.subplots()
+
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.BoundaryNorm([1,3,6,8], cmap.N)
+        coll = plot_point_collection(ax, self.points, self.values, norm = norm)
+        fig.canvas.draw_idle()
+
+        actual_lim = coll.get_clim()
+        assert actual_lim == (1,8)
+        ax.cla()
+
+        fig, ax = plt.subplots()
+
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.BoundaryNorm([1,3,6,8], cmap.N)
+        coll = plot_point_collection(ax, self.points[0:3], self.values[0:3], norm = norm)
+        fig.canvas.draw_idle()
+
+        actual_lim1 = coll.get_clim()
+        assert actual_lim1 == actual_lim
+        ax.cla()
+
+class TestCmapPolygonPlotting:
+    def setup_method(self):
+        # Intializing the dataframe
+        self.N = 10
+        self.points = GeoSeries(Point(i, i) for i in range(self.N)).buffer(0.5)
+
+        values = np.arange(self.N)
+        levels = ['1','1','2','4','5','1','5','4','3','2']
+
+        self.df = GeoDataFrame({'geometry': self.points, 'values': values, 'levels': levels})
+
+    def test_customnorm(self):
+        # test color without additional parameters
+        ax = self.df.plot(column='values')
+        actual_color = ax.collections[0].get_facecolors()
+        cmap = ax.collections[0].get_cmap()
+        expected_color = cmap(np.arange(self.N) / (self.N - 1))
+        np.testing.assert_array_equal(actual_color, expected_color)
+
+        # test color with providing norm
+        norm = mpl.colors.Normalize(vmin = 0, vmax = 12)
+        ax = self.df.plot(column='values', norm = norm)
```

Comment:
as above, you really should be passing in a cmap if you're going to test the colors, but getting the cmap from ax.collections is better practice.